server/coordinator/procedure/drop_table.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metaservicepb"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator/eventdispatch"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventDropTablePrepare = "EventDropTablePrepare"
+	eventDropTableFailed  = "EventDropTableFailed"
+	eventDropTableSuccess = "EventDropTableSuccess"
+
+	stateDropTableBegin   = "StateDropTableBegin"
+	stateDropTableWaiting = "StateDropTableWaiting"
+	stateDropTableFinish  = "StateDropTableFinish"
+	stateDropTableFailed  = "StateDropTableFailed"
+)
+
+var (
+	dropTableEvents = fsm.Events{
+		{Name: eventDropTablePrepare, Src: []string{stateDropTableBegin}, Dst: stateDropTableWaiting},
+		{Name: eventDropTableSuccess, Src: []string{stateDropTableWaiting}, Dst: stateDropTableFinish},
+		{Name: eventDropTableFailed, Src: []string{stateDropTableWaiting}, Dst: stateDropTableFailed},
+	}
+	dropTableCallbacks = fsm.Callbacks{
+		eventDropTablePrepare: dropTablePrepareCallback,
+		eventDropTableFailed:  dropTableFailedCallback,
+		eventDropTableSuccess: dropTableSuccessCallback,
+	}
+)
+
+func dropTablePrepareCallback(event *fsm.Event) {
+	request := event.Args[0].(*DropTableCallbackRequest)
+	table, exists, err := request.cluster.GetTable(request.ctx, request.schemaName, request.tableName)
+	if err != nil {
+		cancelEventWithLog(event, err, "cluster get table")
+		return
+	}
+	if !exists {
+		return
```

Comment:
```suggestion
                 log.warn("drop table stop because table not found",zap.uint64("tableName",request.tableName))
		return
```

Reply:
If the table does not exist, `DropTableProcedure` runs successfully.
I don't think there is a need to print the log here.

Reply:
Dropping a non-existent table may be a corner case when running, and logging for that may helps debugging if any problems happen.